multilinear_extensions/src/mle.rs

```diff
@@ -753,239 +625,289 @@ impl<E: ExtensionField> MultilinearExtension<E> for DenseMultilinearExtension<E>
                 self.evaluations = b;
             }
             (a, b) => panic!(
-                "do not support merge differnt field type DME a: {:?} b: {:?}",
+                "do not support merging different field types: a = {:?}, b = {:?}",
                 a, b
             ),
         }
     }
 
-    /// get ranged multiliear extention
-    fn get_ranged_mle(
+    /// get mle with only subslice range
+    pub fn as_view(&self) -> MultilinearExtension<'_, E> {
+        self.as_view_subslice(1, 0)
+    }
+
+    /// get mle with only subslice range
+    pub fn as_view_subslice(
         &self,
         num_range: usize,
         range_index: usize,
-    ) -> RangedMultilinearExtension<'_, E> {
+    ) -> MultilinearExtension<'_, E> {
         assert!(num_range > 0);
-        let offset = self.evaluations.len() / num_range;
+        let total_len = self.evaluations.len();
+        let offset = total_len / num_range;
```

Comment:
nitpick: shall we call it `slice_len` instead of `offset`? And I think `slice_len = total_len.div_ceil(num_range)`.

Reply:
Do we want to assert that `total_len` is divisible by `num_range`?